quarto/Group_10_Analysis.qmd

```diff
@@ -0,0 +1,281 @@
+knit_print(my_skim(IMDB ))
+```
+
+```{r}
+#| echo: true
+#| warning: false
+
+# Calculate the correlation coefficient between numeric variables
+
+# Filter out non-numeric columns
+numeric_df <- df[sapply(df, is.numeric)]
+
+# Compute correlation matrix
+correlation_matrix <- cor(numeric_df, use = "complete.obs")
+
+# Creating correlation heatmap
+corrplot(cor(IMDB), method = "color", 
+         type = "lower", addCoef.col = 'grey')
+```
+
+```{r}
+#| echo: true
+#| warning: false
+
+# Creating scatterplot between rating and explanatory variables
+
+# Rating vs. Year
+p1 <- ggplot(df, aes(x = year, y = rating, 
+                     color = rating_higher_than_7)) +
+  geom_point() +
+  geom_hline(yintercept = 7, linetype = "dashed", color = "red") +
+  theme_minimal() +
+  theme(legend.position = "none") +
+  labs(title = "Rating vs Year", x = "Year", y = "Rating")
+
+# Rating vs. Length
+p2 <- ggplot(df, aes(x = length, y = rating, 
+                     color = rating_higher_than_7)) +
+  geom_point() +
+  geom_hline(yintercept = 7, linetype = "dashed", color = "red") +
+  theme_minimal() +
+  theme(legend.position = "none") +
+  labs(title = "Rating vs Length", x = "Length", y = "Rating")
+
+# Rating vs. Budget
+p3 <- ggplot(df, aes(x = budget, y = rating, 
+                     color = rating_higher_than_7)) +
+  geom_point() +
+  geom_hline(yintercept = 7, linetype = "dashed", color = "red") +
+  theme_minimal() +
+  theme(legend.position = "none") +
+  labs(title = "Rating vs Budget", x = "Budget", y = "Rating")
+
+# Rating vs. Votes
+p4 <- ggplot(df, aes(x = votes, y = rating, 
+                     color = rating_higher_than_7)) +
+  geom_point() +
+  geom_hline(yintercept = 7, linetype = "dashed", color = "red") +
+  theme_minimal() +
+  theme(legend.position = "none") +
+  labs(title = "Rating vs Votes", x = "Votes", y = "Rating")
+
+# Plot in Grid
+grid.arrange(p1, p2, p3, p4, ncol=2)
+```
+
+```{r}
+#| echo: true
+#| warning: false
+
+# Creating boxplot and barplot between rating_higher_than_7 and explanatory variables
+
+# Rating > 7 vs. Year
+p1 <- ggplot(data = df, mapping = aes(x = factor(rating_higher_than_7), y = year)) +
+  geom_boxplot(fill = "steelblue") +
+  labs(y = "Year", x = "Rating > 7")
+
+# Rating > 7 vs. Length
+p2 <- ggplot(data = df, mapping = aes(x = factor(rating_higher_than_7), y = length)) +
+  geom_boxplot(fill = "steelblue") +
+  labs(y = "Length", x = "Rating > 7")
+
+# Rating > 7 vs. Budget
+p3 <- ggplot(data = df, mapping = aes(x = factor(rating_higher_than_7), y = budget)) +
+  geom_boxplot(fill = "steelblue") +
+  labs(y = "Budget", x = "Rating > 7")
+
+# Rating > 7 vs. Votes
+p4 <- ggplot(data = df, mapping = aes(x = factor(rating_higher_than_7), y = votes)) +
+  geom_boxplot(fill = "steelblue") +
+  labs(y = "Votes", x = "Rating > 7")
+
+# Rating > 7 vs. Genre
+p6 <- ggplot(df, aes(x= genre,  y = ..prop.., group=rating_higher_than_7, fill=rating_higher_than_7)) +
```

Comment:
1. Adjusting the codes (Incomplete pdf output)
2. Adding the legend for barplot